### PR TITLE
Fix OS meta criteria

### PR DESCRIPTION
**Pull request by stonebuzz**
Meta criteria for OS no longer work

GLPI tries to make the relation (LEFT JOIN) from ```glpi_computers_items``` instead of ```glpi_items_operatingsystems```

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | Internal 22385

### CODE REVIEW

File: inc/search.class.php [hunk 1]
```diff
@@ -5388,6 +5388,24 @@ static function addMetaLeftJoin($from_type, $to_type, array &$already_link_table
          return $JOIN;
       }
 
+      if ($from_referencetype === 'Computer' && $to_type === 'OperatingSystem') {
```
Comment (trasher) [inline inc/search.class.php:5391]: Why only computer? Better to rely on presence in `$CFG_GLPI['operatingsystem_types']` I gues..

**Review by trasher — CHANGES_REQUESTED** [review]

File: inc/search.class.php (hunk 1 above)
Comment (stonebuzz) [reply inc/search.class.php]: Good point, I didn't commit my final version.
Fixed

**Review by trasher — COMMENTED** [review]
Sounds good to me. Do you think you can add a unit test on this case?

**Comment by stonebuzz** [comment]
> Sounds good to me. Do you think you can add a unit test on this case?

no problem, I'm planning this =)

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Comment by stonebuzz** [comment]
see https://github.com/glpi-project/glpi/pull/9378

**Pull request closed without merging**